src/helpers.c

```diff
@@ -1250,3 +1251,63 @@ void warn_nosigcheck(const char *file)
 	warn("\n");
 	warn("THE SIGNATURE OF %s WILL NOT BE VERIFIED\n\n", file);
 }
+
+int write_into_file(char *file, char *content, char *mode)
+{
+	FILE *fp = NULL;
+	int ret = 0;
+
+	fp = fopen(file, mode);
+	if (!fp) {
+		return -errno;
+	}
+
+	if (fwrite(content, sizeof(char), sizeof(content), fp) <= 0) {
+		ret = -1;
+		error("There was an error writing to file %s\n", file);
+	}
+
+	fclose(fp);
+	return ret;
+}
+
+int map_file_into_memory(const char *file, unsigned char **buffer, size_t *file_length)
```

Comment:
I would move the map functions to sys too.

Reply:
Why unsigned char?

Reply:
> Why unsigned char?

that is a good question, it was like that in https://github.com/clearlinux/swupd-client/blob/master/src/signature.c#L270, so I just copied it like that, I didn't understand the reason but decided to let it that way

Reply:
You can use char here then